### ISSUE TITLE
The S3 loader doesn't load images from private bucket

### ISSUE THREAD

**Issue by chladog** [cshum/imagor#358]
I can't seem to be able to setup a S3 loader from the bucket that is private, I'm always getting _Forbidden_ from Imagor endpoint.
I have been able to use private bucket for _result storage_ using the S3_RESULT_STORAGE_ACL option, however there is no ACL option for the _loader_.
My motivation is that I do not want to have original files on the S3 public, as I want to allow users to fetch downsized and webp of the images only.
Is there any way to set this up, am I doing something wrong?
I'm using minio s3.

My config
```
            - name: S3_FORCE_PATH_STYLE
              value: '1'
            - name: AWS_REGION
              value: eu
            - name: AWS_ACCESS_KEY_ID
              value: XXXXXXXXXXXXX
            - name: AWS_SECRET_ACCESS_KEY
              value: YYYYYYYYYYYYYY
            - name: S3_ENDPOINT
              value: s3.myminio.com // I tried as well address incl. https protocol
            - name: S3_LOADER_BUCKET
              value: gallery
            - name: S3_RESULT_STORAGE_ACL
              value: readwrite
            - name: S3_RESULT_STORAGE_BUCKET
              value: gallery-results
            - name: PORT
              value: '8000'
            - name: IMAGOR_UNSAFE
              value: '1'
            - name: DEBUG
              value: '1'
            - name: IMAGOR_AUTO_WEBP
              value: '1'
```
And then trying to fetch the image
https://imagorinstance/unsafe/filters:hue(10)/s3.myminio.com/gallery/20190919_103343.jpg

It seems to me it's always trying to fetch it by http loader instead. If I set bucket to public or give it anonymous read policy it works.

**Comment by chladog** [comment]
This is actually working, ty! 

However I found out why this wasn't working for me when I tried it before. I was accessing an image that has diacritics or spaces in the name. For example I have name of the image ```kořen.png```

```https://imagorinstance/unsafe/filters:hue(10)/kořen.png```
```https://imagorinstance/unsafe/filters:hue(10)/ko%C5%99en.png```

I tried both - as is and _encodeURIComponent_. Both giving me:
```
{
  "message": "dial tcp: lookup xn--koen-c5a.png on XX.XXX.XXX.XXX:53: no such host: https://kořen.png",
  "status": 404
}
```

ironically when I have **public bucket** and go ```https://imagorinstance/unsafe/filters:hue(10)/s3.myminio.com/gallery/kořen.jpg```
then it actually works.

So I believe there is a bug in S3 loader not being able to load UTF-8 chars even when encodeURIComponent is used and the HTTP loader is able to load it.

**Comment by cshum** [comment]
imagor follows s3 guidance https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html#object-key-guidelines-safe-characters escaping all special characters except safe characters. The whitelisting is for security concerns.

If you wish to include space as s3 safe characters, you may do so with `S3_SAFE_CHARS`: 
```
S3_SAFE_CHARS="ř"
```

**Comment by chladog** [comment]
I tried to set the env var as advised and still got the error mention above.

However I found a solution of **storing** the encoded-URI as object key "ko%C5%99en.png" while setting S3_SAFE_CHARS="%". And that works fine, but since the guide advice against using "%" character I will follow the safety advice and I'll rename all files to MD5 hashes instead and store the original filename in metadata.

Thank you for your great work!
